### DISCUSSION TITLE
How do I render text into a rectangle of a specified size?

### DISCUSSION THREAD

**Discussion by w4ngzhen** [emilk/egui#3961] [Q&A] [answered]
Hello everyone, I'm writing a small program, is there any way to render text into a specified rectangle?

**Comment by YgorSouza** [comment]
To render text there is the [Label](https://docs.rs/egui/0.25.0/egui/widgets/struct.Label.html) widget.

To constrain the size of the widget, there is the [Ui::add_sized](https://docs.rs/egui/0.25.0/egui/struct.Ui.html#method.add_sized) method.

To specify the size and the position on the screen (instead of just placing it right after the preceding widget), there's the [Ui::put](https://docs.rs/egui/0.25.0/egui/struct.Ui.html#method.put) method.

If the text does not fit in the rectangle, you can use [Label::wrap](https://docs.rs/egui/0.25.0/egui/widgets/struct.Label.html#method.wrap) and [Label::truncate](https://docs.rs/egui/0.25.0/egui/widgets/struct.Label.html#method.truncate) to define how it should be rendered.

**Comment by emilk** [comment, marked as answer]
Use https://docs.rs/egui/latest/egui/struct.Painter.html#method.text with `Align2::CENTER_CENTER` and `rect.center`
